src/consensus/tx_verify.cpp

```diff
@@ -655,10 +655,16 @@ bool Consensus::CheckTxAssets(const CTransaction& tx, CValidationState& state, c
         i++;
         bool fIsAsset = false;
         int nType = 0;
+        int nScriptType = 0;
         bool fIsOwner = false;
-        if (txout.scriptPubKey.IsAssetScript(nType, fIsOwner))
+        if (txout.scriptPubKey.IsAssetScript(nType, nScriptType, fIsOwner))
             fIsAsset = true;
 
+        if (fIsAsset && nScriptType == TX_SCRIPTHASH) {
+            if (!AreP2SHAssetsAllowed())
+                return state.DoS(0, false, REJECT_INVALID, "bad-txns-p2sh-assets-not-active");
```

Comment:
Might be a good idea to move the DoS messages from choose to `BRConstants.h`, although it looks like most of the old ones are hard-coded already. Maybe a separate update.